gitlab/client.py

```diff
@@ -769,36 +775,74 @@ def should_retry() -> bool:
                 if max_retries == -1 or cur_retries < max_retries:
                     wait_time = 2**cur_retries * 0.1
-                    if "Retry-After" in result.headers:
-                        wait_time = int(result.headers["Retry-After"])
-                    elif "RateLimit-Reset" in result.headers:
-                        wait_time = int(result.headers["RateLimit-Reset"]) - time.time()
+                    if "Retry-After" in backend_response.headers:
+                        wait_time = int(backend_response.headers["Retry-After"])
+                    elif "RateLimit-Reset" in backend_response.headers:
+                        wait_time = (
+                            int(backend_response.headers["RateLimit-Reset"])
+                            - time.time()
+                        )
                     cur_retries += 1
                     time.sleep(wait_time)
                     continue
 
-            error_message = result.content
+            error_message = backend_response.content
             try:
-                error_json = result.json()
+                error_json = backend_response.json()
                 for k in ("message", "error"):
                     if k in error_json:
                         error_message = error_json[k]
             except (KeyError, ValueError, TypeError):
                 pass
 
-            if result.status_code == 401:
+            if backend_response.status_code == 401:
                 raise gitlab.exceptions.GitlabAuthenticationError(
-                    response_code=result.status_code,
+                    response_code=backend_response.status_code,
                     error_message=error_message,
-                    response_body=result.content,
+                    response_body=backend_response.content,
                 )
 
             raise gitlab.exceptions.GitlabHttpError(
-                response_code=result.status_code,
+                response_code=backend_response.status_code,
                 error_message=error_message,
-                response_body=result.content,
+                response_body=backend_response.content,
             )
 
+    def http_request(
+        self,
+        verb: str,
+        path: str,
+        query_data: Optional[Dict[str, Any]] = None,
+        post_data: Optional[Union[Dict[str, Any], bytes]] = None,
+        raw: bool = False,
+        streamed: bool = False,
+        files: Optional[Dict[str, Any]] = None,
+        timeout: Optional[float] = None,
+        obey_rate_limit: bool = True,
+        retry_transient_errors: Optional[bool] = None,
+        max_retries: int = 10,
+        **kwargs: Any,
+    ) -> requests.Response:
+        utils.warn(
+            "`http_request()` is deprecated and will be removed in a future version.\n"
+            "Please use `backend_request()` instead.",
+            category=DeprecationWarning,
+        )
```

Comment:
I'm a bit wary of this, it would feel wrong deprecating this after we already had a similar discussion in https://github.com/python-gitlab/python-gitlab/pull/1842. What would be the issue with having a `http_request` in the public client?

Keeping in mind that the user does not care about the backends generally.

Reply:
>This is a breaking change, so this would need to be at least communicated to users via a breaking change trailer (and triger 4.0.0).

I've thought it would make sense using this method and not to trigger 4.0.0 for now.

Reply:
@lmilbaum what I meant was what would be wrong with simply keeping `http_backend` on the client? It's quite consistent with the existing `http_*` methods (there is some history in the other MR, where we discussed we'd keep it in the public client as it is).

Reply:
I am not following. `http_backend` is kept on the client while having backend-specific implementation in the backend package.
I was just trying to address the valid concern you have raised regarding the breaking change.